advisor/src/main/kotlin/Advisor.kt

```diff
@@ -67,9 +70,21 @@ class Advisor(
 
         val providers = providerFactories.map { it.create(config) }
 
+        val results = sortedMapOf<Identifier, List<AdvisorResult>>()
+
         val packages = ortResult.getPackages(skipExcluded).map { it.pkg }
-        val results = runBlocking { providers.first().retrievePackageVulnerabilities(packages) }
-            .mapKeysTo(sortedMapOf()) { (pkg, _) -> pkg.id }
+        runBlocking {
+            providers.map { provider ->
+                async {
+                    provider.retrievePackageVulnerabilities(packages)
+                }
+            }.forEach { providerResults ->
+                providerResults.await().forEach { (pkg, advisorResults) ->
+                    val resultsForId = results.getOrDefault(pkg.id, emptyList())
```

Comment:
Get and update of the `results` map can probably be simplified by using Map's `merge()` method.